projects/core/src/lib/datatable/table/table.component.ts

```diff
@@ -131,11 +132,37 @@ export class DataTableComponent implements OnInit, AfterViewInit, OnDestroy, OnC
         this.dataSource.data = data.data;
         this.featureColumns = metadata.dataTypes.keySeq().toArray();
         this.featureColumnDataTypes = metadata.dataTypes.valueSeq().toArray();
-        this.displayedColumns = ['_____select', '_____coordinates', '_____table__start', '_____table__end'].concat(this.featureColumns);
+        if (this.displayedColumns.length === 0) {
+            this.displayedColumns = ['_____select', '_____coordinates', '_____table__start', '_____table__end'].concat(this.featureColumns);
+        }
         this.featureColumnDataTypes = this.getColumnProperties();
+        this.checkboxLabels = ['_____coordinates', '_____table__start', '_____table__end'].concat(this.featureColumns);
         setTimeout(() => this.navigatePage(this.projectService.getSelectedFeature()));
     }
 
+    showCheckboxColumn(): void {
```

Comment:
please add some info what this does and maybe use a more descriptive name

Reply:
I changed the name and added some info above the method